### PR TITLE
API for CRUD Operations

### PR DESCRIPTION
**Pull request by Vijeta28**
API for CRUD Operations

### CODE REVIEW

File: src/main/java/com/rathore/DentalAdminServiceApplication.java [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
 public class DentalAdminServiceApplication {
 
-	public static void main(String[] args) {
-		SpringApplication.run(DentalAdminServiceApplication.class, args);
-	}
+    public static void main(String[] args) {
+        SpringApplication.run(DentalAdminServiceApplication.class, args);
+    }
 
+    @Bean
```
Comment (vijay-rathore93) [inline src/main/java/com/rathore/DentalAdminServiceApplication.java]: remove from here and put in configuration file

File: src/main/java/com/rathore/configuration/Config.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		return new ModelMapper();
+	}
+
+	@Bean
+	public PasswordEncoder passwordEncoder() {
+		return new BCryptPasswordEncoder(10);
+	}
+
+	@Bean
+	public Docket api() {
```
Comment (vijay-rathore93) [inline src/main/java/com/rathore/configuration/Config.java]: Some more information has to put like license, version, etc

File: src/main/java/com/rathore/controller/AdminController.java [hunk 3]
```diff
@@ -1,5 +1,56 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+import static com.rathore.utility.ApplicationResource.USER_BASE_URL;
+
+@RestController
```
Comment (vijay-rathore93) [inline src/main/java/com/rathore/controller/AdminController.java]: Swagger is not implemented properly in controllers

Add error code and success code also

File: src/main/java/com/rathore/entity/Role.java [hunk 4]
```diff
@@ -1,5 +1,12 @@
 package com.rathore.entity;
 
-public class Role {
```
Comment (vijay-rathore93) [inline src/main/java/com/rathore/entity/Role.java]: No swagger no annotation

File: src/main/java/com/rathore/entity/User.java [hunk 5]
```diff
@@ -13,63 +13,24 @@
...
 import javax.persistence.Transient;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
 import org.springframework.security.core.GrantedAuthority;
 import org.springframework.security.core.userdetails.UserDetails;
 
 @Entity
 @Table(name = "User")
-public class User implements UserDetails {
```
Comment (vijay-rathore93) [inline src/main/java/com/rathore/entity/User.java]: No swagger configuration for properties wise

like @APIProperty annotation for property description

File: src/main/java/com/rathore/exceptions/InvalidData.java [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package com.rathore.exceptions;
+
+public class InvalidData {
```
Comment (vijay-rathore93) [inline src/main/java/com/rathore/exceptions/InvalidData.java]: Create an Exception properly

File: src/main/java/com/rathore/exceptions/UserNotFoundException.java [hunk 7]
```diff
@@ -0,0 +1,4 @@
+package com.rathore.exceptions;
+
+public class UserNotFoundException {
```
Comment (vijay-rathore93) [inline src/main/java/com/rathore/exceptions/UserNotFoundException.java]: Delete this Spring already has  UserNotFound Exception

File: src/main/java/com/rathore/model/ResponseDTO.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class ResponseDTO {
+
```
Comment (vijay-rathore93) [inline src/main/java/com/rathore/model/ResponseDTO.java]: No Swagger Configuration

File: src/main/java/com/rathore/model/UserDTO.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
+package com.rathore.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class UserDTO {
```
Comment (vijay-rathore93) [inline src/main/java/com/rathore/model/UserDTO.java]: No Swagger Configuration

File: src/main/java/com/rathore/DentalAdminServiceApplication.java (hunk 1 above)
Comment (Vijeta28) [reply src/main/java/com/rathore/DentalAdminServiceApplication.java]: Done

File: src/main/java/com/rathore/exceptions/UserNotFoundException.java (hunk 7 above)
Comment (Vijeta28) [reply src/main/java/com/rathore/exceptions/UserNotFoundException.java]: done

File: src/main/java/com/rathore/exceptions/InvalidData.java (hunk 6 above)
Comment (Vijeta28) [reply src/main/java/com/rathore/exceptions/InvalidData.java]: Done

File: src/main/java/com/rathore/entity/User.java (hunk 5 above)
Comment (Vijeta28) [reply src/main/java/com/rathore/entity/User.java]: Done

File: src/main/java/com/rathore/exceptions/InvalidData.java (hunk 6 above)
Comment (Vijeta28) [reply src/main/java/com/rathore/exceptions/InvalidData.java]: Done

File: src/main/java/com/rathore/model/ResponseDTO.java (hunk 8 above)
Comment (Vijeta28) [reply src/main/java/com/rathore/model/ResponseDTO.java]: Done

File: src/main/java/com/rathore/model/UserDTO.java (hunk 9 above)
Comment (Vijeta28) [reply src/main/java/com/rathore/model/UserDTO.java]: - Done

File: src/main/java/com/rathore/configuration/Config.java (hunk 2 above)
Comment (Vijeta28) [reply src/main/java/com/rathore/configuration/Config.java]: need to be discuss

**Pull request merged**